package-res/resources/web/pentaho/visual/base/model.js

```diff
@@ -45,37 +73,287 @@ define([
+         * or due to an invalid selection mode) yields an error [result]{@link pentaho.lang.ActionResult}
+         * and lead to the emission (and processing) of the
+         * ["rejected:select"]{@link pentaho.visual.base.events.RejectedSelect}.
+         * In this case, this method returns a [result]{@link pentaho.lang.ActionResult}
+         * with an exception in the `error` field.
+         *
+         * @param {!pentaho.data.filter.AbstractFilter} proposedDataFilter - A filter representing
+         * the data set which will be used to modify the current selection.
+         * @param {?object} keyArgs - Keyword arguments.
+         * @param {!function} keyArgs.selectionMode - A function that computes a new selection,
+         * taking into account the current selection and an input `dataFilter`.
+         * @return {pentaho.lang.ActionResult}
+         * If unsuccessful, the `error` field contains the exception that originated the error.
+         * If successful,  the `error` field is `null` and the `value` field contains the updated current selection.
+         *
+         * @fires "will:select"
+         * @fires "did:select"
+         * @fires "rejected:select"
+         *
+         * @see #_runPipeline
+         * @see #_changeSelection
+         * @see #_broadcastResult
+         * @see pentaho.visual.base.types.selectionModes
+         */
+        selectAction: function(proposedDataFilter, keyArgs) {
+          var selectionMode = O.getOwn(keyArgs, "selectionMode");
+          var dataFilter = proposedDataFilter;
+
+          var result = this._runPipeline(WillSelect, RejectedSelect, dataFilter, selectionMode);
+          if(result) {
+            if(result.error) return result;
+            selectionMode = result.value.selectionMode;
+            dataFilter = result.value.dataFilter;
+          }
+
+          try {
+            result = this._changeSelection(dataFilter, selectionMode);
+          } catch(e) {
+            result = new ActionResult(undefined, e);
+          }
+          return this._broadcastResult(DidSelect, RejectedSelect, result, dataFilter);
+        },
+
+        /**
+         * Updates the current selection based on a filter and on a selection mode.
+         *
+         * This method is invoked by [selectAction]{@link pentaho.visual.base.Model#selectAction} once all
+         * listeners of the ["will:select"]{@link pentaho.visual.base.events.WillSelect} event have eventually
+         * mutated the original input filter and the selection mode to be used.
+         *
+         * @param {!pentaho.data.filter.AbstractFilter} proposedDataFilter -
+         * @param {!function} proposedSelectionMode -
+         * @return {pentaho.lang.ActionResult}
+         * If unsuccessful, the `error` field contains the exception that originated the error.
+         * If successful,  the `error` field is `null` and the `value` field contains the updated current selection.
+         *
+         * @fires "will:change:select"
```

Comment:
Not sure this should be documented here, as this is part of a sub-action. Perhaps mentioning that the set is performed and sending the user there for possible events?